model/openapi.yml

```diff
@@ -557,6 +557,7 @@ components:
             type: string
             enum:
               - CPU
+              - CONN
```

Comment:
From the discussion in https://github.com/elastic/elastic-agent/issues/4880, we want to have the HTTP connection request diagnostics as an optional value that is enabled by default.